plugin/trino-iceberg/src/main/java/io/trino/plugin/iceberg/catalog/glue/TrinoGlueCatalog.java

```diff
@@ -356,16 +357,54 @@ public Table loadTable(ConnectorSession session, SchemaTableName table)
     @Override
     public void dropTable(ConnectorSession session, SchemaTableName schemaTableName)
     {
-        BaseTable table = (BaseTable) loadTable(session, schemaTableName);
-        validateTableCanBeDropped(table);
+        com.amazonaws.services.glue.model.Table metastoreTable;
```

Comment:
Why do we need any changes to the Drop table logic in a catalog implementation?

perhaps, dropping corrupted table could be done separately, as an unregister_table + delete table directory.
if the table is corrupted, we probably should not expect catalog do anything smarter than that

cc @alexjo2144 @findinpath @electrum thoughts?

Reply:
> unregister_table + delete table directory.

I like this approach. I was concerned when `data files of the Iceberg table may be located in different locations`, But when there is the corrupted table (metadata file is missing) then we won't able to perform `dropTableData` because we won't able to load the table.

So I think it is fine to use `unregister_table + delete table directory.`.

```java
        // Use the Iceberg routine for dropping the table data because the data files
        // of the Iceberg table may be located in different locations
        dropTableData(table.io(), table.operations().current());
```

Reply:
> perhaps, dropping corrupted table could be done separately, as an unregister_table + delete table directory.

Do we have the certainty in such cases that the directory can be safely removed  or should this operation be left to the system administrator?
IMO we are speaking here about a corner case which should probably be handled in a best effort manner.

I'd advocate to simply unregister the table.